docs/architecture/blazor-for-web-forms-developers/migration.md

```diff
@@ -1,15 +1,646 @@
+        <a href="@($"/Catalog/Edit/{_item.Id}")" class="esh-link-item">
+            Edit
+        </a>
+        |
+        <a href="/" class="esh-link-item">
+            Back to list
+        </a>
+    </div>
+
+</div>
+
+@code {
+    private CatalogItem _item;
+
+    [Parameter]
+    public int Id { get; set; }
+
+    protected override void OnInitialized()
+    {
+        Logger.LogInformation("Now loading... /Catalog/Details/{Id}", Id);
+
+        _item = CatalogService.FindCatalogItem(Id);
+    }
+}
+```
+
+Notice that the code and markup are in the same file. Any required services are made accessible with the `@inject` attribute. Per the `@page` directive, this page can be accessed at the `Catalog/Details/{id}` route. The value of the route's `{id}` placeholder has been constrained to an integer. As described in the [routing](pages-routing.md) section, unlike Web Forms, a Razor component explicitly states its route and any parameters that are included. Many Web Forms controls may not have exact counterparts in Blazor. There's often an equivalent HTML snippet that will serve the same purpose. For example, the `<asp:Label />` control can be replaced with an HTML `<label>` element.
+
+### Model validation in Blazor
+
+If your Web Forms code includes validation, you can transfer much of what you have with little-to-no changes. A benefit to running in Blazor is that the same validation logic can be run without needing custom JavaScript. Data annotations enable easy model validation.
+
+For example, the *Create.aspx* page has a data entry form with validation. An example snippet would look like this:
+
+```aspx
+<div class="form-group">
+    <label class="control-label col-md-2">Name</label>
+    <div class="col-md-3">
+        <asp:TextBox ID="Name" runat="server" CssClass="form-control"></asp:TextBox>
+        <asp:RequiredFieldValidator runat="server" ControlToValidate="Name" Display="Dynamic"
+            CssClass="field-validation-valid text-danger" ErrorMessage="The Name field is required." />
+    </div>
+</div>
+```
+
+In Blazor, the equivalent markup is provided in a *Create.razor* file:
+
+```razor
+<EditForm Model="_item" OnValidSubmit="@...">
+    <DataAnnotationsValidator />
+
+    <div class="form-group">
+        <label class="control-label col-md-2">Name</label>
+        <div class="col-md-3">
+            <InputText class="form-control" @bind-Value="_item.Name" />
+            <ValidationMessage For="(() => _item.Name)" />
+        </div>
+    </div>
+    
+    ...
+</EditForm>
+```
+
+The `EditForm` context includes validation support and can be wrapped around input. Data annotations are a common way to add validation. Such validation support can be added via the `DataAnnotationsValidator` component. For more information on this mechanism, see [ASP.NET Core Blazor forms and validation](/aspnet/core/blazor/forms-validation).
+
+## Migrate built-in Web Forms controls
+
+[ TBD ]
```

Comment:
@twsouthwick What's the plan for this section? Should we leave it in for now?

Reply:
@danroth27 We have to gather a bunch of data for this section that we don't have. Probably should leave it in with a comment that it will be filled in before release (not sure if we already have boilerplate text for that).